docs/performance.md

```diff
@@ -0,0 +1,53 @@
+# Performance
+
+## Rerenders
+Duplicate rerenders of React components is a common
+performance bottleneck in React Native apps.
+
+Try to follow these tips to avoid needless rerenders of your components.
+
+1. Divide your components into Presentational and Container components. This helps in separation of concerns and also increase performance.
+Refer to this [document](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0).
+
+2. Use [PureComponent](https://facebook.github.io/react/docs/react-api.html#react.purecomponent) where props and component level state is not mutated.
+
+3. Always declare handler functions outside your render method. This prevents new references to your function from being passed every time your component receives props, for example:
+
+    prefer
+    ```javascript
+        handleOnPress() {
+            ...do something
+        }
+
+        render() {
+            return (
+                <WrappedComponent onPress={this.handleOnPress} />
+            );
+        }
+    ```
+
+    over
+    ```javascript
+        render() {
+            return (
+                <WrappedComponent onPress={() => {...do something}} />
+            );
+        }
+    ```
+
+4. Avoid using `…this.props`
+Passing all the props to wrapped components is generally not needed and requires more shallow comparisons than required (when using PureComponent).
+
+5. When connecting to redux store, narrow your request as much as possible, for example:
+
+    prefer `state.app.isHydrated` over `state.app`.
+
+6. Avoid using props that will change frequently like scrollOffset
+
+7. In general, try to have primitive props, which are immutable and will make switching to PureComponents easier.
+
```

Comment:
I would add another: 'make sure to not create new objects on each render when passing to children, this will prevent PureComponent optimziation because the two object will be referentially different'